### PR TITLE
Wanl/fix fist preview version bump

### PR DESCRIPTION
**Pull request by wanlwanl**
# Issue
#8522 

# Problem
The bug is introduced by new codegen [pr](https://github.com/Azure/autorest.typescript/commit/b0853b2bd11a6fa5ec49543e7212d148b16bc8b7). The way to define default api version is changed.
extracting the api version should also consider version of codegen.

e.g. new api-version at client level is defined [here](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/mongocluster/arm-mongocluster/src/rest/documentDBClient.ts#L23)
while old one is:
- [here](https://github.com/Azure/azure-sdk-for-js/blob/06716722818f5838cc10a9e5644b7ba9f32089d5/sdk/face/ai-vision-face-rest/src/faceClient.ts#L26)
- [another](https://github.com/Azure/azure-sdk-for-js/blob/06716722818f5838cc10a9e5644b7ba9f32089d5/sdk/openai/openai-rest/src/openAIClient.ts#L22)

# Solution
1. try to detect in new client, return api-version if found
2. fallback to detect old client, return whatever found

### CODE REVIEW

File: tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts [hunk 1]
```diff
@@ -1,81 +1,135 @@
...
     return apiVersions;
-}
+};
 
-const getApiVersionTypeFromOperations: IApiVersionTypeExtractor = (packageRoot: string): ApiVersionType => {
-    const paraPath = path.join(packageRoot, 'src/rest/parameters.ts');
+const getApiVersionTypeFromOperations: IApiVersionTypeExtractor = (
+    packageRoot: string
+): ApiVersionType => {
+    const paraPath = path.join(packageRoot, "src/rest/parameters.ts");
```
Comment (qiaozha) [inline tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts:115]: same here, wonder if we should leverage this function for standalone rlc?

**Review by qiaozha — APPROVED** [review]
LGTM, some minor comments

File: tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts [hunk 2]
```diff
@@ -1,81 +1,135 @@
...
 
-import { ApiVersionType } from "../../common/types"
+import { ApiVersionType } from "../../common/types";
 import { IApiVersionTypeExtractor } from "../../common/interfaces";
 import { getTsSourceFile } from "../../common/utils";
+import { readFileSync } from "fs";
 
 const findRestClientPath = (packageRoot: string): string => {
-    const restPath = path.join(packageRoot, 'src/rest/');
+    const restPath = path.join(packageRoot, "src/rest/");
```
[the remark this answers is not in the document]
Comment (wanlwanl) [reply tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts:12]: this function in 'mlc' folder, means it only design for modular client. but we can make the function more general to cover both modular and RLC.

File: tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts (hunk 1 above)
Comment (wanlwanl) [reply tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts:115]: this function in 'mlc' folder, means it only design for modular client. but we can make the function more general to cover both modular and RLC.

File: tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts [hunk 3]
```diff
@@ -1,81 +1,135 @@
...
+): ApiVersionType => {
     const clientPath = findRestClientPath(packageRoot);
     const apiVersion = findApiVersionInRestClient(clientPath);
-    if (apiVersion && apiVersion.indexOf("-preview") >= 0) return ApiVersionType.Preview;
-    if (apiVersion && apiVersion.indexOf("-preview") < 0) return ApiVersionType.Stable;
+    if (apiVersion && apiVersion.indexOf("-preview") >= 0)
+        return ApiVersionType.Preview;
+    if (apiVersion && apiVersion.indexOf("-preview") < 0)
+        return ApiVersionType.Stable;
     return ApiVersionType.None;
```
[the remark this answers is not in the document]
Comment (wanlwanl) [reply tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts:87]: Thanks for the reminder, let's discuss when we developing this tool for unbranding users.

**Pull request merged**